File: cpp/Program/Parkinglot.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"
#include <vector>

using namespace std;

class Parkinglot
{
private:
    int kapasitas;
    int jumKendaraan;
    std::vector<Vehicle> vehicles;


public:
    Parkinglot(){

    }

    Parkinglot(int kapasitas){
        this->kapasitas = kapasitas;
        jumKendaraan = 0;
        this->vehicles = {};
    }
    Parkinglot(int kapasitas, int jumKendaraan){
        this->kapasitas = kapasitas;
        this->jumKendaraan = jumKendaraan;
        this->vehicles = {};
    }
    Parkinglot(int kapasitas, int jumKendaraan, std::vector<Vehicle> vehicles){
        this->kapasitas = kapasitas;
        this->jumKendaraan = jumKendaraan;
        vehicles = vehicles;
    }

    int GetKapasitas()  {
        return kapasitas;
    }

    void SetKapasitas(int kapasitas) {
        this->kapasitas = kapasitas;
    }

    int GetJumKendaraan()  {
        return jumKendaraan;
    }

    void SetJumKendaraan(int jumKendaraan) {
        this->jumKendaraan = jumKendaraan;
    }

    std::vector<Vehicle> GetVehicles()  {
        return vehicles;
    }

    void SetVehicles(std::vector<Vehicle> vehicles) {
        vehicles = vehicles;
    }

    void parkVehicle(Vehicle v) {
        if (jumKendaraan < kapasitas) {
            vehicles.push_back(v);
            jumKendaraan++;
        } else {
            std::cout << "Parking lot is full." << std::endl;
        }
    }

    ~Parkinglot(){

    }
};

File: python/Program/ParkingLot.py
from Vehicle import Vehicle

class ParkingLot:
    __kapasitas = ''
    __jumKendaraan = ''
    
    
    
    def __init__(self, kapasitas , jumKendaraan):
        self.__kapasitas = kapasitas
        self.__jumKendaraan = jumKendaraan
        self.__kendaraan = []
 
    def get_Kendaraan(self):
        return self.__kendaraan
    
    def set_Kendaraan(self, value):
        if not self.__kendaraan:  # Jika daftar kendaraan kosong
            self.__kendaraan.append(Vehicle(platNomor = "", merk= "", tahunProduksi= "", warna= ""))  # Tambahkan kendaraan kosong
        self.__kendaraan.append(value)  # Tambahkan kendaraan ke daftar

    def get_kapasitas(self):
        return self.__kapasitas

    def set_kapasitas(self, value):
        self.__kapasitas = value

    def get_jumKendaraan(self):
        return self.__jumKendaraan

    def set_jumKendaraan(self, value):
        self.__jumKendaraan = value

    